src/library/dlgtrackinfo.cpp

```diff
@@ -395,13 +393,25 @@ void DlgTrackInfo::saveTrack() {
         updatedRows.insert(oldRow);
 
         QVariant vHotcue = hotcueItem->data(Qt::DisplayRole);
-        if (vHotcue.canConvert<int>()) {
-            int iTableHotcue = vHotcue.toInt();
-            // The GUI shows hotcues as 1-indexed, but they are actually
-            // 0-indexed, so subtract 1
-            pCue->setHotCue(iTableHotcue - 1);
+        bool ok;
+
+
+        int iTableHotcue = vHotcue.toInt(&ok);
+        if (ok) {
+            if (iTableHotcue == 0 && !theCueFound) {
+                // adopt the first 0 hot cue as the cue
+                pCue->setHotCue(-1);
+                pCue->setType(Cue::LOAD);
+                theCueFound = true;
+            } else {
+                // The GUI shows hotcues as 1-indexed, but they are actually
+                // 0-indexed, so subtract 1
+                pCue->setHotCue(iTableHotcue - 1);
+                pCue->setType(Cue::CUE);
+            }
         } else {
             pCue->setHotCue(-1);
+            pCue->setType(Cue::CUE);
```

Comment:
A hot cue with an invalid number? What are the consequences? Can we add a comment about when this occurs and what we are doing here and why?